mapwmslayer.c

```diff
@@ -874,11 +886,25 @@ msBuildWMSLayerURL(mapObj *map, layerObj *lp, int nRequestType,
 
     pszExceptionsParam = msOWSLookupMetadata(&(lp->metadata),
                          "MO", "exceptions_format");
-    if (pszExceptionsParam == NULL) {
-      if (nVersion >= OWS_1_1_0 && nVersion < OWS_1_3_0)
-        pszExceptionsParam = "application/vnd.ogc.se_inimage";
+      
+    if (!bIsEssential) {
+      if (pszExceptionsParam == NULL) {
+        if (nVersion >= OWS_1_1_0 && nVersion < OWS_1_3_0)
+          pszExceptionsParam = "application/vnd.ogc.se_inimage";
+        else
+          pszExceptionsParam = "INIMAGE";
+      }
+    }
+    /* if layer is essential, set EXCEPTIONS parameter to XML */    
+    else {                        
+      if (nVersion >= OWS_1_3_0)
```

Comment:
I've not reviewed earlier WMS specs, but at least for WMS 1.1.1 and 1.3.0, the default value for EXCEPTIONS when it is not specified is to be in XML format. So a simpler possibility, that would save if else loginc on the WMS version, would be not to emit EXCEPTIONS= at all.

Reply:
This same check occurs earlier in the file.

Reply:
This pull request follows your advice in #6061 where you gave 2 options for the solution.

Reply:
Sorry for making it complicated, I definitely tried.  I have tested this on multiple systems; that said, I'm open to changes/new pull request/closing this and re-opening...

Reply:
no need to close&recreate a pull request to update it. I've just added a commit with the code simplification I add in mind. Can you pull and confirm this works OK ?

Reply:
confirmed, your simplification still returns the XML exception (instead of image).  Thank you again @rouault